test/distributed/test_c10d_nccl.py

```diff
@@ -2796,6 +2796,7 @@ def test_pass_nccl_options_config(self):
         pg_opts.config.min_ctas = 2
         pg_opts.config.cga_cluster_size = 2
         pg_opts.config.net_name = "Socket"
+        pg_opts.config.split_share = 1
```

Comment:
Not necessary to do it in this PR. Do you think we want to add some docs in distributed_c10d.py about NCCL config in the option? Currently the only info we have is by reading through cpp code and unit test, IIUC.

Reply:
more clearer doc is always welcome ^_^

Reply:
but at least we should put a comment saying what this does?  i'm actually not clear on when i would change it-

is it this? or something else?
1= share resources so less memory and sockets are used (unclear how much savings here vs the tradeoff of 0)
0= can overlap p2p comms but waste more communicators and init time?

Reply:
All those are NCCL specific configs, which are documented by NCCL as the source of truth and referred in the Pybind file (init.cpp) line 2739: https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/api/types.html#ncclconfig-t. 
<img width="831" alt="image" src="https://github.com/user-attachments/assets/17e73179-835a-41a3-8af2-8a0f734f9f93">. Specifically, for this config, it explains as "Specify whether to share resources with child communicator during communicator split. Set the value of splitShare to 0 or 1. The default value is 0. When the parent communicator is created with splitShare=1 during ncclCommInitRankConfig, the child communicator can share internal resources of the parent during communicator split. Split communicators are in the same family. When resources are shared, aborting any communicator can result in other communicators in the same family becoming unusable. Irrespective of whether sharing resources or not, users should always abort/destroy all no longer needed communicators to free up resources.

"